controllers/quay/quayregistry_controller.go

```diff
@@ -120,10 +120,108 @@ func (r *QuayRegistryReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, nil
 	}
 
-	created := v1.GetCondition(quay.Status.Conditions, v1.ConditionComponentsCreated)
-	if created != nil && created.Reason == v1.ConditionReasonMigrationsInProgress {
-		log.Info("migrations in progress, skipping reconcile")
-		return ctrl.Result{}, nil
+	created := v1.GetCondition(updatedQuay.Status.Conditions, v1.ConditionComponentsCreated)
+	migrationInProgress := created != nil && created.Reason == v1.ConditionReasonMigrationsInProgress
+	if migrationInProgress {
+		log.Info("checking Quay upgrade `Job` completion")
+
+		var upgradeJob batchv1.Job
+		err := r.Client.Get(
+			ctx,
+			types.NamespacedName{
+				Name:      updatedQuay.GetName() + "-" + v1.QuayUpgradeJobName,
+				Namespace: updatedQuay.GetNamespace(),
+			},
+			&upgradeJob,
+		)
+
+		// similarly to when a v1.ConditionReasonMigrationsFailed occurs,
+		// when the upgrade job is expected to exist but doesn't
+		// (i.e. someone manually removed it) we want the reconcile loop
+		// to run in its entirety, so we change the condition reason to
+		// something other than migrations in progress.
+		if err != nil && errors.IsNotFound(err) {
+			_, err = r.updateWithCondition(
+				updatedQuay,
+				v1.ConditionComponentsCreated,
+				metav1.ConditionFalse,
+				v1.ConditionReasonMigrationsJobMissing,
+				"upgrade job not found")
+			if err != nil {
+				log.Error(err, "failed to update `conditions` of `QuayRegistry`")
+			}
+			return ctrl.Result{RequeueAfter: time.Second}, nil
+		}
+
+		if err != nil {
+			log.Error(err, "could not retrieve Quay upgrade `Job`")
+			return ctrl.Result{RequeueAfter: time.Second}, nil
+		}
+
+		if upgradeJob.Status.Active == 1 {
+			return ctrl.Result{RequeueAfter: time.Second}, nil
```

Comment:
It could be useful to log something here. We can see that the loop is being called but awaiting for the migration to finish. wdyt?

Reply:
yup, will add an info.